android/app/src/main/java/DeviceInfo/Utils.java

```diff
@@ -178,9 +180,11 @@ public static String getOurVersion(Context ctx) {
 
 
 
+    public static String parseJSON(String json, String peerID) throws JSONException {
+        
+        JSONObject obj = new JSONObject(json);
+        obj.put("fromUID", peerID);
 
-    public static int suma(int a, int b){
-        return a + b;
+        return  obj.toString();
```

Comment:
This is small enough to put it on `onNewMessage` :wink:

Reply:
but,This little piece of code has nothing to do with the event, that's why it is separate hehe